libsolidity/formal/SMTEncoder.cpp

```diff
@@ -779,7 +779,16 @@ void SMTEncoder::visitTypeConversion(FunctionCall const& _funCall)
 	auto argument = _funCall.arguments().front();
 	unsigned argSize = argument->annotation().type->storageBytes();
 	unsigned castSize = _funCall.annotation().type->storageBytes();
-	if (argSize == castSize)
+	// Allow same-size casts as well as literal arguments.
+	// TODO: support StringLiteral
+	if (auto category = argument->annotation().type->category(); category == Type::Category::RationalNumber)
+	{
+		auto const& numberType = dynamic_cast<RationalNumberType const&>(*argument->annotation().type);
```

Comment:
Could assert that `typemin <= literalValue <= typemax`

Reply:
Why should we do that when it is a constant? It just seems extra assertions for z3 to deal with.

If this value is converted again, then it will have the limit assertions. If it is reassigned, then I guess it will have it too if it is coming from a non-literal?

Reply:
Ah yes, no need to

Reply:
My worry is that something like `uint8 x = 100000000` would cause weird things for the solver, since `x` would have type restrictions and `100000000` would be out of that range. But Solidity already forbids it, so it's fine.